### PR TITLE
Fix `listDatabases` filtering when using `nameOnly`

### LINKED ISSUE #3624: `listDatabases` filtering is broken with `nameOnly` options
### What should be done?

TBD

### Where?

TBD

### Definition of Done

- all handlers updated;
- unit tests added/updated;
- integration/compatibility tests added/updated;
- spot refactorings done;
- user documentation updated or an issue to create documentation created;
- something else?

### PR DESCRIPTION
**Pull request by henvic**
# Description

This fix restores the filtering behavior before 6131bf4d21c0dcd20ddab63c219cf8c6a4541b44.

Closes #3624.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by henvic** [comment]
Still working on tests.

**Comment by henvic** [comment]
I am currently checking existing compatibility tests as I shall add them yet.

File: internal/backends/collection_test.go [hunk 1]
```diff
@@ -266,6 +266,45 @@ func TestCollectionUpdateAll(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
+			t.Run("DatabaseWasFiltered", func(t *testing.T) {
```
Comment (AlekSi) [inline internal/backends/collection_test.go:269]: Let's remove that test

File: integration/commands_administration_test.go [hunk 2]
```diff
@@ -159,24 +160,112 @@ func TestCommandsAdministrationListDatabases(t *testing.T) {
...
+		assert.NotZero(t, actual.TotalSize, "TotalSize should be non-zero")
+	})
+
+	t.Run("ExistsNameOnly", func(t *testing.T) {
+		actual, err := db.Client().ListDatabases(ctx, bson.D{
+			{Key: "name", Value: name},
+		},
+			&options.ListDatabasesOptions{
+				NameOnly: pointer.To(true),
+			})
```
Comment (AlekSi) [inline integration/commands_administration_test.go:190]: That formatting is unlike anything we already use. Let's use `options.ListCollections().SetNameOnly(true)` – we use methods for options everywhere

File: integration/commands_administration_test.go [hunk 3]
```diff
@@ -159,24 +160,112 @@ func TestCommandsAdministrationListDatabases(t *testing.T) {
 	db := collection.Database()
 	name := db.Name()
 
-	actual, err := db.Client().ListDatabases(ctx, bson.D{{"name", name}})
-	require.NoError(t, err)
-	require.Len(t, actual.Databases, 1)
+	t.Run("Exists", func(t *testing.T) {
+		actual, err := db.Client().ListDatabases(ctx, bson.D{{Key: "name", Value: name}})
```
Comment (AlekSi) [inline integration/commands_administration_test.go:164]:
```suggestion
		t.Parallel()

		actual, err := db.Client().ListDatabases(ctx, bson.D{{Key: "name", Value: name}})
```

Subtests should run in parallel too

Comment (henvic) [reply integration/commands_administration_test.go:165]: Added t.Parallel() call.

File: integration/commands_administration_test.go [hunk 4]
```diff
@@ -159,24 +159,138 @@ func TestCommandsAdministrationListDatabases(t *testing.T) {
 	db := collection.Database()
 	name := db.Name()
 
-	actual, err := db.Client().ListDatabases(ctx, bson.D{{"name", name}})
-	require.NoError(t, err)
-	require.Len(t, actual.Databases, 1)
+	testCases := []struct { //nolint:vet // for readability
```
Comment (noisersup) [inline integration/commands_administration_test.go:162]: I'd suggest to make a map here to stay consistent with our other integration tests (such as this https://github.com/FerretDB/FerretDB/blob/main/integration/create_test.go#L252-L259)

**Review by noisersup — APPROVED** [review]
Apart from minor comment above, looks great to me!

File: integration/commands_administration_test.go (hunk 4 above)
Comment (henvic) [reply integration/commands_administration_test.go]: Sure! Changed to a map.

File: integration/commands_administration_test.go [hunk 5]
```diff
@@ -159,24 +159,137 @@ func TestCommandsAdministrationListDatabases(t *testing.T) {
...
-			SizeOnDisk: actual.Databases[0].SizeOnDisk,
-			Empty:      actual.Databases[0].Empty,
-		}},
-		TotalSize: actual.TotalSize,
+	testCases := map[string]struct { //nolint:vet // for readability
+		filter any
+		opts   []*options.ListDatabasesOptions
+
+		empty    bool
+		zeroSize bool
```
[the remark this answers is not in the document]
Comment (henvic) [reply integration/commands_administration_test.go]: I've simplified and removed zeroSize, dbCount, and empty. But added a expectedNameOnly bool.

File: integration/commands_administration_test.go [hunk 6]
```diff
@@ -159,24 +159,137 @@ func TestCommandsAdministrationListDatabases(t *testing.T) {
...
+		dbCount  int
```
[the remark this answers is not in the document]
Comment (henvic) [reply integration/commands_administration_test.go]: Fixed. I'm checking now how to add an extra test with multiple databases.

**Pull request merged**